projects/coordinator/src/main/java/org/batfish/coordinator/VersionCompatibilityFilter.java

```diff
@@ -0,0 +1,42 @@
+package org.batfish.coordinator;
+
+import com.google.common.base.Strings;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.batfish.common.CoordConsts;
+import org.batfish.common.Version;
+
+/**
+ * This filter verifies that the client's version is compatible with the version of Batfish service.
+ */
+public class VersionCompatibilityFilter implements ContainerRequestFilter {
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) {
+    String clientVersion = requestContext.getHeaderString(CoordConsts.SVC_KEY_VERSION);
+    if (Strings.isNullOrEmpty(clientVersion)) {
+      requestContext.abortWith(
+          Response.status(Status.PRECONDITION_FAILED)
```

Comment:
Not sure which status code to use here, seems like '426 Upgrade Required' specifically refers to the protocol in use, not the client version.